apps/dashboard/app/views/files/_shell.html.erb

```diff
@@ -0,0 +1,17 @@
+<div class="shell-wrapper btn-group dropright">
+  <%= link_to OodAppkit.shell.url(path: @path.to_s).to_s, id: 'open-in-terminal-btn', class: 'btn btn-outline-dark btn-sm disabled', target: '_blank' do %>
+    <i class="fas fa-terminal" aria-hidden="true"></i>
+    Open in Terminal
+  <% end %>
+
+  <% if OodAppkit.clusters.count > 0 %>
```

Comment:
https://github.com/OSC/ondemand/blob/f27530a3f0eb927886053520b98fce9fcd3e0842/apps/dashboard/app/apps/ood_app.rb#L87-L116

We don't want all clusters, only

```
      login_clusters = OodCore::Clusters.new(
        OodAppkit.clusters
          .select(&:allow?)
          .reject { |c| c.metadata.hidden }
          .select(&:login_allow?)
      )
```

Reply:
That should be a method. Maybe Configuration.login_clusters?

Reply:
ApplicationHelper knows this (I just have it fresh i my mind from activejobs and OODClusters).

https://github.com/OSC/ondemand/blob/f27530a3f0eb927886053520b98fce9fcd3e0842/apps/dashboard/app/helpers/application_helper.rb#L6-L8

Of course that doesn't reject hidden.  I guess I'd have to lookup where this is used and if it should.